code/game/objects/structures/window.dm

```diff
@@ -209,67 +210,97 @@ var/global/wcCommon = pick(list("#379963", "#0d8395", "#58b5c3", "#49e46e", "#8f
+			if(state >= 1)
+				if(takes_time)
+					to_chat(user, "<span class='notice'>You begin to [(state == 1) ? "fasten the window to" : "unfasten the window from"] the frame.</span>")
+					if(!do_after(user, 20 * W.toolspeed, target = src))
+						return 1
+				state = 3 - state
+				to_chat(user, (state == 1 ? "<span class='notice'>You have unfastened the window from the frame.</span>" : "<span class='notice'>You have fastened the window to the frame.</span>"))
+		else
+			if(takes_time)
+				to_chat(user, "<span class='notice'>You begin to [anchored ? "unfasten the frame from" : "fasten the frame to"] the floor.</span>")
+				if(!do_after(user, 20 * W.toolspeed, target = src))
+					return 1
 			anchored = !anchored
 			update_nearby_icons()
-			playsound(loc, I.usesound, 75, 1)
 			to_chat(user, (anchored ? "<span class='notice'>You have fastened the window to the floor.</span>" : "<span class='notice'>You have unfastened the window.</span>"))
-	else if(istype(I, /obj/item/weapon/crowbar) && reinf && state <= 1)
+		return 1
+	//crowbar
+	if(iscrowbar(W))
+		if(!reinf || state > 1)
+			return 0
+		playsound(loc, W.usesound, 75, 1)
+		if(takes_time)
+			to_chat(user, "<span class='notice'>You begin to pry the window [state ? "out of" : "in to"] the frame.</span>")
+			if(!do_after(user, 20 * W.toolspeed, target = src))
+				return 1
 		state = 1 - state
-		playsound(loc, I.usesound, 75, 1)
 		to_chat(user, (state ? "<span class='notice'>You have pried the window into the frame.</span>" : "<span class='notice'>You have pried the window out of the frame.</span>"))
-	else if(istype(I, /obj/item/weapon/wrench) && !anchored && health > 7) //Disassemble deconstructed window into parts
-		playsound(src.loc, I.usesound, 50, 1)
+		return 1
+	//wrench
+	if(iswrench(W))
+		if(anchored)
+			return 0
+		playsound(loc, W.usesound, 50, 1)
+		if(takes_time)
+			to_chat(user, "<span class='notice'>You begin to disassemble [src]...</span>")
+			if(!do_after(user, 20 * W.toolspeed, target = src))
+				return 1
 		for(var/i=0;i<sheets;i++)
-			var/obj/item/stack/sheet/glass/NG = new glasstype(src.loc)
-			for(var/obj/item/stack/sheet/glass/G in src.loc) //Stack em up
-				if(G==NG)
+			var/obj/item/stack/sheet/NS = new glasstype(get_turf(src))	//glass types don't share a base tye of /glass, so this didn't work for plasma glass
+			for(var/obj/item/stack/sheet/S in loc) //Stack em up
+				if(S==NS)
 					continue
-				if(G.amount>=G.max_amount)
+				if(S.amount >= S.max_amount)
 					continue
-				G.attackby(NG, user, params)
+				S.attackby(NS, user)
 
 			if(reinf)
-				var/obj/item/stack/rods/NR = new (src.loc)
-				for(var/obj/item/stack/rods/R in src.loc)
+				var/obj/item/stack/rods/NR = new (get_turf(src))
+				for(var/obj/item/stack/rods/R in loc)
 					if(R==NR)
 						continue
-					if(R.amount>=R.max_amount)
+					if(R.amount >= R.max_amount)
 						continue
-					R.attackby(NR, user, params)
+					R.attackby(NR, user)

```

Comment:
This should still pass params, even if they're unused- MarkVA's original commit made sure of this.